Source/ui_unix/GamePad/GamePadUtils.cpp

```diff
@@ -0,0 +1,48 @@
+#include "GamePadUtils.h"
+#include <cstring>
+
+std::array<uint32, 6> CGamePadUtils::GetDeviceID(libevdev *dev)
+{
+	std::array<uint32, 6> device{0};
+	if (libevdev_get_uniq(dev) != NULL)
+	{
+		if(!CGamePadUtils::ParseMAC(device, libevdev_get_uniq(dev)))
+		{
+			const char* tmp_id = libevdev_get_uniq(dev);
+			if(strlen(tmp_id) >= 6)
+			{
+				for(int i = 0; i < strlen(tmp_id) && i < 6; ++i)
+					device.at(i) = tmp_id[i];
+			}
+		}
+	}
+	if(device == std::array<uint32, 6>{0})
+	{
+		int vendor = libevdev_get_id_vendor(dev);
+		int product = libevdev_get_id_product(dev);
+		int ver = libevdev_get_id_version(dev);
```

Comment:
Using these 3 id's can result in a non unique ID (if you have 2 of the same controller), but gives the user the freedom to plug the controller into any port on their PC.
Using `libevdev_get_id_bustype` this will give a unique id but you'd lose that freedom.